src/Parser.purs

```diff
@@ -0,0 +1,64 @@
+module Parser (Node, PseudoAst(..), parseFromFile) where
+
+import Prelude
+import Data.Either (Either)
+import Data.Foldable (fold)
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Show (genericShow)
+import Data.Maybe (maybe)
+import Data.Newtype (class Newtype)
+import Data.String.CodeUnits (singleton)
+import Data.String.Extra (camelCase)
+import Data.Traversable (sequence)
+import Effect.Class (class MonadEffect, liftEffect)
+import Foreign.Generic (class Decode, defaultOptions, genericDecode)
+import Node.Encoding (Encoding(..))
+import Node.FS.Sync (readTextFile)
+import Node.Path (FilePath)
+import Text.Parsing.StringParser (ParseError, runParser)
+import Text.Parsing.StringParser.CodeUnits (alphaNum, char, string)
+import Text.Parsing.StringParser.Combinators (choice, many1, optionMaybe)
+import Utils (getClassNames)
+
+type Node
+  = { className :: String, name :: String }
+
+newtype PseudoAst
+  = PseudoAst (Array Node)
+
+derive newtype instance eqPseudoAst :: Eq PseudoAst
+
+derive newtype instance ordPseudoAst :: Ord PseudoAst
+
+derive instance newtypePseudoAst :: Newtype PseudoAst _
+
+derive instance genericPseudoAst :: Generic PseudoAst _
+
+instance showPseudoAst :: Show PseudoAst where
+  show = genericShow
+
+instance decodePseudoAst :: Decode PseudoAst where
+  decode = genericDecode defaultOptions { unwrapSingleConstructors = true }
+
+nodeFromClassName :: String -> Either ParseError Node
+nodeFromClassName className = ({ className, name: _ }) <$> camelCase <$> runParser nameParser className
+  where
+  nameParser = ado
+    neg <- maybe "" (const "neg-") <$> optionMaybe (char '-')
+    name <- fold <$> many1 anyValidAndSpecialChar
+    in neg <> name
+
+  anyValidAndSpecialChar =
+    choice
+      [ const "-over-" <$> string "\\\\/"
+      , const "-" <$> string "\\\\:"
+      , singleton <$> alphaNum
+      , string "\\:"
+      , string "-"
+      , string "\\/"
+      ]
```

Comment:
How is it different if we just return this in Js:
```
return [...acc, className];
```
And here it will be:
```
  anyValidAndSpecialChar =
    choice
      [ const "-over-" <$> string "\\/"
      , const "-" <$> string "\\:"
      , singleton <$> alphaNum
      ]
``` 
I'm pretty sure I'm missing something, but just out of curiousity xD

Reply:
Unfortunately we need to escape \ at one point, otherwise when printed by Handlebars we'll lose the \ and the string will sometimes be invalid ("\:" is not a valid string while "\\:" is).
So at one point I had to add some \ logic 😞 
That being said, that logic could be moved in PureScript, we just need a ClassNameParse that would replace the occurrences lf \ by \\

Also, i looked for an option in Handlebars but couldn'tt fond any

Reply:
But for the last 3 parsers it's just a copy paste mistake xD
Let me removed it ^^
(I'd like to remove the references to AppM they're not really useful and quite confusing 😕)